txrelayer/txrelayer.go

```diff
@@ -130,6 +135,12 @@ func (t *TxRelayerImpl) sendTransactionLocked(txn *ethgo.Transaction, key ethgo.
 		return ethgo.ZeroHash, err
 	}
 
+	if t.writer != nil {
+		_, _ = t.writer.Write([]byte(
+			fmt.Sprintf("[ROOTCHAIN - CONTRACTS DEPLOYMENT TRANSACTION DETAIL]\nFrom = %s \nGas = %d  \nGas Price = %d\n",
```

Comment:
This is wrong. You should not have hardcoded logging this kind of message here, because `TxRelayer` is used not only by `rootchain deploy` command, but on multiple places.

I feel that you should provide some kind of logging callback, instead of explicitly hardcoding logging message here.

Reply:
Can the logging message be just "Transaction detail" ?

Reply:
Or we can add a message argument to the WithWriter function.

Reply:
> Can the logging message be just "Transaction detail" ?

...or you can simply add [TxRelayer.SendTransaction] as prefix and then transaction details. Although check with @Nemanja0x, about this point (https://github.com/0xPolygon/polygon-edge/pull/1627#discussion_r1232097459). However you find it better.

Reply:
It is not bad idea to have a message argument, but however you decide. Just do not want to to  have hardcoded "ROOTCHAIN ..."